### PR TITLE
ocs-to-ocs: fetch cephFilesystemSubVolumeGroup resources

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit fetch cephFilesystemSubVolumeGroup resources
from rook and append it to the external resource list.

Signed-off-by: subhamkrai <srai@redhat.com>

### CODE REVIEW

**Comment by subhamkrai** [comment]
on top of #1492

File: services/provider/server/server.go [hunk 1]
```diff
@@ -334,6 +336,40 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
...
+			provisionerCephClientSecret, err := s.getCephClientSecretName(ctx, i.CephClients["provisioner"])
+			if err != nil {
+				return nil, err
+			}
+
+			extR = append(extR, &pb.ExternalResource{
+				Name: "cephfs",
+				Kind: "StorageClass",
+				Data: mustMarshal(map[string]string{
+					"clusterID": hash(clusterID),
```
Comment (nb-ohad) [inline services/provider/server/server.go:361]: Why are we splitting the creation of the cluster id into two places? (here and Line 345)
Can we just do it here, or there?

File: services/provider/server/server.go [hunk 2]
```diff
@@ -334,6 +336,40 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 					"csi.storage.k8s.io/node-stage-secret-name":        nodeCephClientSecret,
 					"csi.storage.k8s.io/controller-expand-secret-name": provisionerCephClientSecret,
 				})})
+		case "CephFilesystemSubVolumeGroup":
+			subVolumeGroup := &rookCephv1.CephFilesystemSubVolumeGroup{}
+			err := s.client.Get(ctx, types.NamespacedName{Name: i.Name, Namespace: s.namespace}, subVolumeGroup)
+			if err != nil {
+				return nil, fmt.Errorf("failed to get %s cephFilesystemSubVolumeGroup. %v", i.Name, err)
+			}
+			clusterID := fmt.Sprintf("%s-%s-file-%s", subVolumeGroup.Namespace, subVolumeGroup.Spec.FilesystemName, subVolumeGroup.Name)
```
Comment (nb-ohad) [inline services/provider/server/server.go:345]: This is defined very far for the place it is used, any reason to do so?

File: services/provider/server/server.go [hunk 3]
```diff
@@ -363,3 +399,8 @@ func mustMarshal(data map[string]string) []byte {
 	}
 	return newData
 }
+
+func hash(s string) string {
+	h := sha256.Sum256([]byte(s))
+	return hex.EncodeToString(h[:16])
+}
```
Comment (nb-ohad) [inline services/provider/server/server.go:406]: Why do we need a util function to a 2 liner that is used only once? Can't we just embed this in the call site?

File: services/provider/server/server.go [hunk 4]
```diff
@@ -334,6 +336,40 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
...
+
+			extR = append(extR, &pb.ExternalResource{
+				Name: "cephfs",
+				Kind: "StorageClass",
+				Data: mustMarshal(map[string]string{
+					"clusterID": hash(clusterID),
+					"csi.storage.k8s.io/provisioner-secret-name":       provisionerCephClientSecret,
+					"csi.storage.k8s.io/node-stage-secret-name":        nodeCephClientSecret,
+					"csi.storage.k8s.io/controller-expand-secret-name": provisionerCephClientSecret,
+				})})
```
Comment (nb-ohad) [inline services/provider/server/server.go:365]: Why are we splitting the creation of the cluster id into two places? (here and Line 345)
Can we just do it here, or there?

Comment (nb-ohad) [inline services/provider/server/server.go:365]: A suggestion related to my comment on the hash util (below)

File: services/provider/server/server.go (hunk 1 above)
Comment (nb-ohad) [inline services/provider/server/server.go:361]: A suggestion related to my comment on the hash util (below)

**Pull request merged**